src/gobeventproducer/database/local/contextmanager.py

```diff
@@ -0,0 +1,57 @@
+from sqlalchemy import create_engine
+from sqlalchemy.engine.url import URL
+from sqlalchemy.orm import Session
+
+from gobeventproducer.config import DATABASE_CONFIG
+from gobeventproducer.database.local.model import Base, LastSentEvent
+
+
+class LocalDatabaseConnection:
+    """Abstraction for receiving and updating the last event that is sent."""
+
+    def __init__(self, catalogue: str, collection: str):
+        self.catalogue = catalogue
+        self.collection = collection
+        self.session = None
+        self.last_event = None
+
+    def _connect(self):
+        engine = create_engine(URL(**DATABASE_CONFIG), connect_args={"sslmode": "require"})
+        Base.metadata.bind = engine
+        self.session = Session(engine)
+
+    def __enter__(self):
+        """Enter context, connect to local database."""
+        self._connect()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Exit context, commit last event to database."""
+        self.session.commit()
```

Comment:
Do we need to always commit here, in case of exception too?

Reply:
I'd say yes. Short version: The contextmanager is just saving last eventids to the database. Whenever an eventid is updated, this always happened after sending the corresponding event. So if it fails somehow, it's safe to commit this to database.

Long version (but in short): there are probably exceptions to find to the logic above, but I'd opt for keeping it simple in this case. Just commit the stuff to database. Hopefully in the future we won't even need the database anymore; currently it's only used for catching any potential unexpected mismatches in events produced earlier vs what we need to produce now. If no mismatches occur, it's safe to say we can do without later :)